broker/core/src/main/groovy/com/swisscom/cloud/sb/broker/backup/shield/ShieldApiException.java

```diff
@@ -0,0 +1,36 @@
+package com.swisscom.cloud.sb.broker.backup.shield;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.client.HttpStatusCodeException;
+
+public class ShieldApiException extends RuntimeException {
+    private final String description;
+    private final HttpStatus httpStatus;
+
+    private ShieldApiException(String description, Throwable ex) {
+        super(description, ex);
+        this.description = description;
+        httpStatus = (ex instanceof HttpStatusCodeException) ? ((HttpStatusCodeException) ex).getStatusCode(): null;
+    }
+
+    public static ShieldApiException of(String description, Throwable ex) {
+        return new ShieldApiException(description, ex);
+    }
+
+    public boolean isHttpStatusCodeException() {
+        return httpStatus != null;
+    }
+
+    public boolean isNotFound() {
+        return isHttpStatusCodeException() && (httpStatus.value() == 404) || httpStatus.value() == 501;
```

Comment:
```suggestion
        return isHttpStatusCodeException() && (httpStatus.value() == 404 || httpStatus.value() == 501);
```
Is what I think what you wanted (but a switch case would be better for adding easily more cases in the future).

What I don't like is that according to this method an HTTP status code of 501 is a _not found_. A _not implemented_ is not the same (if I check the `isNotFound()` and decide to create the resource in that case and receive a _not implemented_ again?).

If it is a specific case of Shield, is a perfect candidate for being explained in the javadoc. So, you decide: document it or change it.